Add basic Rocket list classes.

alesrazym opened the pull request.

alesrazym committed:
Add basic Rocket list classes.

Create `RocketActivity` with `RocketListFragment` and its adapter, view model and repo. Also add icon assets for list item.

alesrazym commented:
@havlisimo I've added basic classes, as discussed in person.

Will continue with `Koin`, probably in a separate branch so that we can close PRs.

havlisimo commented on app/src/main/java/cz/quanti/razym/rocketapp/domain/RocketsRepo.kt:
```diff
@@ -0,0 +1,5 @@
+package cz.quanti.razym.rocketapp.domain
+
+interface RocketsRepo {
```
mozna jsem to blbe napsal na ten papir, ale pouzivame cele slovo Repository pro interface i implementaci

Commit:
REF: Rename `xxxRepo` to `xxxRepository`

Commit:
REF: Rename `xxxRepoImpl` to `xxxRepositoryImpl`

havlisimo commented on app/src/main/res/layout/activity_rocket.xml:
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8"?>
+<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
lepsi je pouzit FragmentContainerView

havlisimo commented on app/src/main/java/cz/quanti/razym/rocketapp/system/RocketListFragment.kt:
```diff
@@ -0,0 +1,41 @@
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        return inflater.inflate(R.layout.fragment_rocket_list, container, false)
```
pouzij view binding misto rucniho inflatovani a pouzivani findViewById
https://developer.android.com/topic/libraries/view-binding

havlisimo commented on app/src/main/java/cz/quanti/razym/rocketapp/domain/RocketModel.kt:
```diff
@@ -0,0 +1,3 @@
+package cz.quanti.razym.rocketapp.domain
```
model by mel byt v package model (to jsem ti asi vcera rekl spatne) a nemel by obsahovat slovo Model, staci Rocket

havlisimo commented on app/src/main/java/cz/quanti/razym/rocketapp/presentation/RocketListAdapter.kt:
```diff
@@ -0,0 +1,39 @@
+package cz.quanti.razym.rocketapp.presentation
```
patry do system vrstvy

havlisimo commented on app/src/main/java/cz/quanti/razym/rocketapp/system/RocketListFragment.kt:
```diff
@@ -0,0 +1,41 @@
+
+        viewModel.rocketLiveData.observe(this) { rockets ->
+            val recycler = requireView().findViewById<RecyclerView>(R.id.rocket_list)
+            recycler.layoutManager = LinearLayoutManager(requireContext())
```
lze nastavit v xml 
        app:layoutManager="androidx.recyclerview.widget.LinearLayoutManager"

havlisimo commented on app/src/main/res/layout/rocket_list_item.xml:
```diff
@@ -0,0 +1,71 @@
+        app:layout_constraintStart_toEndOf="@id/barrier1"
+        tools:text="First Flight 2006-03-24" />
+
+    <androidx.constraintlayout.widget.Barrier
```
obe bariery mi prijdou nadbytecne

ikonu rakety zarovnas podle parenta na start, top, bottom
ikonu chevronu podle parenta na end, top, bottom
a textviews mezi ne

Commit:
REF: Use `FragmentContainerView` instead of `FrameLayout`

Commit:
REF: Use View binding instead of `findViewById()`

Commit:
REF: Correct package usage, set `layoutManager` in xml

Commit:
REF: Remove `Barrier` from `rocket_list_item.xml`

Commit:
REF: Rename `RocketModel` to `Rocket`

Commit:
LIB: Update libs

Update `com.google.android.material:material` to 1.10.0
Change `androidx.fragment:fragment` to `-ktx`
